src/zenml/models/v2/plugin/plugin_flavor.py

```diff
@@ -12,14 +12,43 @@
 #  or implied. See the License for the specific language governing
 #  permissions and limitations under the License.
 """Plugin flavor model definitions."""
-from pydantic import BaseModel
+from typing import Generic, TypeVar
+
+from pydantic import Extra, Field
 
 from zenml.enums import PluginSubType, PluginType
+from zenml.models.v2.base.base import (
+    BaseResponse,
+    BaseResponseMetadata,
+    BaseZenModel,
+)
+
+
+class BasePluginResponseBody(BaseZenModel):
+    """Response body for plugins."""
+
+
+class BasePluginResponseMetadata(BaseResponseMetadata):
+    """Response metadata for plugins."""
 
 
-class BasePluginFlavorResponse(BaseModel):
+AnyPluginBody = TypeVar("AnyPluginBody", bound=BasePluginResponseBody)
+AnyPluginMetadata = TypeVar(
+    "AnyPluginMetadata", bound=BasePluginResponseMetadata
+)
+
+
+class BasePluginFlavorResponse(
+    BaseResponse[AnyPluginBody, AnyPluginMetadata],
+    Generic[AnyPluginBody, AnyPluginMetadata],
+):
     """Base response for all Plugin Flavors."""
 
-    flavor_name: str
-    plugin_type: PluginType
-    plugin_subtype: PluginSubType
+    flavor_name: str = Field(title="Name of the flavor.")
+    plugin_type: PluginType = Field(title="Type of the flavor.")
+    plugin_subtype: PluginSubType = Field(title="Subtype of the flavor.")
+
+    class Config:
+        """Configuration for base plugin flavor response."""
+
+        extra = Extra.allow
```

Comment:
"Allow" means that the plugins can add whatever they want to the core models defined by ZenML and they will be serialized and show up in the responses sent through the REST API. Are you sure you don't mean "ignore" here? "Ignore" will also not raise errors if extra attributes are present, but will silently discard them.

Reply:
problem is the endpoint is decorated with BasePluginFlavorResponse - however the plugin registry returns a girhubwebhookeventsourceflavorresponse, which has extra fields. but i guess this might be soled by moving those into the body as well